ksqldb-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -364,37 +358,33 @@ public SchemaKGroupedStream groupBy(
       final Stacker contextStacker
   ) {
     if (!repartitionNeeded(groupByExpressions)) {
-      return groupByKey(keyFormat.getFormatInfo(), valueFormat, contextStacker);
+      return groupByKey(keyFormat, valueFormat, contextStacker);
     }
 
-    final FormatInfo keyFmtInfo = keyFormat.getFormatInfo().getFormat().equals(NoneFormat.NAME)
-        ? FormatInfo.of(ksqlConfig.getString(KsqlConfig.KSQL_DEFAULT_KEY_FORMAT_CONFIG))
-        : keyFormat.getFormatInfo();
-
-    final KeyFormat rekeyedKeyFormat = KeyFormat
-        .nonWindowed(keyFmtInfo, SerdeFeatures.of());
-
+    final boolean isSingleKey = groupByExpressions.size() == 1;
```

Comment:
If this is removed and we instead hard-code `isSingleKey=true` for now, as we have been for purposes of setting internal key features, and is still the case in this PR for table group bys, do the changes in QTT historic plans go away? If so, I prefer this. I'll re-introduce this change in my multli-column group by PR where plans will change anyway.

Reply:
sounds good, I can do that